api/src/main/java/org/openmrs/api/impl/OrderSetServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.openmrs.OrderSet;
+import org.openmrs.OrderSetMember;
+import org.openmrs.api.APIException;
+import org.openmrs.api.OrderSetService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.OrderSetDAO;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+import java.util.Date;
+import java.util.List;
+
+@Transactional
```

Comment:
Better to avoid doing a class-level @Transactional annotation on the service, and instead put this only on the methods that need it.